packages/react/src/host/controller.tsx

```diff
@@ -1,15 +1,17 @@
 import type {ComponentType} from 'react';
-import type {RemoteComponentType} from '@remote-ui/core';
+import type {Controller, Renderer} from './types';
+
+import {renderComponent} from './RemoteComponent';
+import {renderText} from './RemoteText';
 
 export interface ComponentMapping {
   [key: string]: ComponentType<any>;
 }
 
-export interface Controller {
-  get(type: string | RemoteComponentType<string, any, any>): ComponentType<any>;
-}
-
-export function createController(components: ComponentMapping): Controller {
+export function createController(
+  components: ComponentMapping,
+  renderer: Partial<Renderer> = {},
```

Comment:
I know it's a little annoying from a typing perspective, but IMO it would be nicer if consumers could pass custom `renderComponent`/ `renderText` functions that will get called with the "default" version of those functions. This is a pattern I've seen in a bunch of other libraries where you might want to customize only a subset of operations, but delegate back to the library for the default behavior in other cases.